src/SharedAcceptanceTests.RequirePartitionKey/When_using_synchronized_session_via_container.cs

```diff
@@ -37,31 +37,28 @@ public class Endpoint : EndpointConfigurationBuilder
         {
             public Endpoint()
             {
-                EndpointSetup<DefaultServer>((config, run) =>
-                {
-                    config.UsePersistence<CosmosDbPersistence>().RegisterSharedTransactionalBatchForDependencyInjection();
-                });
+                EndpointSetup<DefaultServer>();
             }
 
             public class MyMessageHandler : IHandleMessages<MyMessage>
             {
-                public MyMessageHandler(TransactionalBatch batch, Context context)
+                public MyMessageHandler(ICosmosDBStorageSession storageSession, Context context)
```

Comment:
This shouldn't have been reverted. If `TransactionalBatch` is constructed by the container, it will not be the same as the transactional batch returned by `ICosmosDBStorageSession`.

Reply:
Alternatively, the test assertion would need to be modified.

Reply:
No the problem is I stopped midway. The other handlers need to be changed according to the change done in this handler and then everything works as intended. The injected session batch has to be the right one

Reply:
Ah, ok. I've changed to do the custom resolution of `ICosmosDBStorageSession` and registering `.Batch` off it, which is equivalent. But could do so that all 3 handlers to be injected with `ICosmosDBStorageSession`.

Reply:
Yeah it was my original intent to show injection of the session because that is the key feature we want to test. I just stopped half way through unfortunately. You know, context switches... Sorry for that

Reply:
Except when we do that, the first and the second handlers will not receive the same session object (tests fail).

Reply:
Confirmed [here](https://github.com/Particular/NServiceBus.Persistence.CosmosDB/pull/41/commits/42137c47b9722da02a4febf35b61a12615f96687), the session is not the same and that causes the test to fail.

Reply:
Doesn't that mean we have a bug? Isn't the synchronised storage session surrounding all the handlers and thus they should be the same for same message? Maybe we haven't seen the bug yet because previously we used uow lifetime which means the batch was only resolved once per pipeline

Reply:
Possibly. Needs to be looked into.